### PR TITLE
Gradient with Momentum, RMSprop, Adam's Optimization Implemented

### PR DESCRIPTION
**Pull request by kidrahahjo**
I'm working on the implementation of optimization algorithms of Deep Neural Network

### CODE REVIEW

**Comment by kidrahahjo** [comment]
can someone let me know why this failed?

**Comment by tirthasheshpatel** [comment]
Not your fault. It is a upstream issue for keras

**Comment by kidrahahjo** [comment]
Then will all the pull requests get declined?

**Comment by tirthasheshpatel** [comment]
Mostly not. Maybe until keras fixes it. (which will take time as they are not very active)

**Comment by kidrahahjo** [comment]
so what should we do if we wish to contribute?

**Comment by tirthasheshpatel** [comment]
Sometimes maintainers take time to merge. But prs such as this do get merged

**Comment by kidrahahjo** [comment]
okay thanks a lot

**Comment by dmeoli** [comment]
Thanks @kidrahahjo. The maintained optimizer code now lives in deep_learning4e.py, which already provides stochastic_gradient_descent and adam. This PR adds momentum/RMSprop/adam to the legacy 3e BackPropagationLearner in learning.py (a beyond-book extension, changing its signature), which is superseded by the 4e implementation. Closing as superseded - thanks for the contribution!
